src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -512,6 +542,9 @@ void queue_actor_task(LocalSchedulerState *state,
     }
   }
 
+  /* Create a new task queue entry. */
+  TaskQueueEntry elt = TaskQueueEntry_init(spec, task_spec_size);
+  insert_actor_task_queue(state, algorithm_state, elt);
```

Comment:
Re-ordering this so that we are guaranteed the task exists in redis when the insertion happens.